src/main/resources/assets/computercraft/shaders/monitor.frag

```diff
@@ -0,0 +1,40 @@
+#version 430 core
+
+#define FONT_WIDTH 6.0
+#define FONT_HEIGHT 9.0
+
+layout (location = 2) uniform sampler2D u_font;
+layout (location = 3) uniform int u_width;
+layout (location = 4) uniform int u_height;
+layout (location = 5) uniform samplerBuffer u_tbo;
+layout (location = 6) uniform vec3 u_palette[16];
+
+in vec2 f_pos;
+
+vec2 texture_corner(int index) {
+    float x = 1.0 + float(index % 16) * (FONT_WIDTH + 2.0);
+    float y = 1.0 + float(index / 16) * (FONT_HEIGHT + 2.0);
+    return vec2(x, y);
+}
+
+void main() {
+    vec2 term_pos = vec2(f_pos.x / FONT_WIDTH, f_pos.y / FONT_HEIGHT);
+    vec2 corner = floor(term_pos);
+
+    ivec2 cell = ivec2(corner);
+    int index = 3 * (clamp(cell.x, 0, u_width - 1) + clamp(cell.y, 0, u_height - 1) * u_width);
+
+    if(cell.x < 0 || cell.y < 0 || cell.x >= u_width || cell.y >= u_height) {
```

Comment:
GPUs aren't good at branching (although getting better). Can we get rid of this somehow? If not, see if we can at least use branching functions like [step](https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/step.xhtml), these are very fast.

Reply:
Mostly as a reminder to myself, but I _think_ `step(x, 0) * step(width - 1, x) * step(y, 0) * step(height - 1, y)` will work. Then we can just multiply `img.a` by that.